modules/assim.sequential/R/load_data_paleon_sda.R

```diff
@@ -0,0 +1,148 @@
+                               x$variable.name
+                             }, 
+                             USE.NAMES = FALSE), 
+                      use.names = FALSE)
+  
+  start_date <- settings$state.data.assimilation$start.date
+  end_date   <- settings$state.data.assimilation$end.date
+  
+  obs.mean <- list()
+  obs.cov <- list()
+  
+  start.time <- format(lubridate::ymd(start_date),settings$state.data.assimilation$forecast.time.step)
+  end.time <- format(lubridate::ymd(end_date),settings$state.data.assimilation$forecast.time.step)
+  obs.times <- start.time:end.time
+  
+  
+  for(i in seq_along(format_id)){
+    input.list[[i]] <- db.query(paste("SELECT * FROM inputs WHERE site_id =",site$id ,"  AND format_id = ",format_id[[i]]), bety$con)
+    input.id[[i]] <- input.list[[i]]$id
+    
+    data.path <- PEcAn.DB::query.file.path(input.id[[i]], settings$host$name, bety$con)
+    format_full <- format <- PEcAn.DB::query.format.vars(input.id = input.id[[i]], bety, format.id = NA, var.ids=NA)
+    
+    format$na.strings <- 'NA'
+    
+    # ---- LOAD INPUT DATA ---- #
+    time.row <- format$time.row
+    vars.used.index <- setdiff(seq_along(format$vars$variable_id), format$time.row)
+    
+    print(paste('Using PEcAn.benchmark::load_data.R on format_id',format_id[[i]],'-- may take a few minutes'))
+    obvs[[i]] <- PEcAn.benchmark::load_data(data.path, format, start_year = lubridate::year(start_date), end_year = lubridate::year(end_date), site, vars.used.index, time.row)
+    
+    dataset <- obvs[[i]]
+    variable <- var.names
+    
+    ### Tree Ring Data Product
+    if(format_id[[i]] == '1000000040'){
+      obvs[[i]] <- obvs[[i]][obvs[[i]]$model_type=='Model RW + Census',]
+      obvs[[i]]$AbvGrndWood <- obvs[[i]]$AbvGrndWood * .48
+      obvs[[i]]$NPP_1_C <- obvs[[i]]$NPP_1_C * .48
+      arguments <- list(.(year, MCMC_iteration, site_id), .(variable))
+      arguments2 <- list(.(year), .(variable))
+      arguments3 <- list(.(MCMC_iteration), .(variable), .(year))
+    }else{
+      print('ERROR: This data format has not been added to this function (ツ)_/¯ ')
+      stop()
+    }
+    
+    ### Map species to model specific PFTs
+    if(any(var.names == 'AGB.pft')){
+      spp_id <- match_species_id(unique(dataset$species_id),format_name = 'usda',bety)
+      spp_id <- spp_id[spp_id$input_code!='HAVI4',]
+      pft_mat <- match_pft(spp_id$bety_species_id, settings$pfts, con = bety$con)
+      
+      x <- paste0('AGB.pft.', pft_mat$pft)
+      names(x) <- spp_id$input_code
+      dataset$pft.cat <- x[dataset$species_id]
+      dataset <- dataset[dataset$pft.cat!='NA_AbvGrndWood',]
+      variable <- sub('AGB.pft','AbvGrndWood',variable)
+      arguments <- list(.(year, MCMC_iteration, site_id, pft.cat), .(variable))
+      arguments2 <- list(.(year, pft.cat), .(variable))
+      arguments3 <- list(.(MCMC_iteration), .(pft.cat, variable), .(year))
+    } 
+    
+    print('Now, melting data')
```

Comment:
1) use a logger function, not print
2) maybe a more descriptive message? What if I don't know what 'melt' is -- could you explain more generally the goal of the next few steps?